protocol/contracts/templegold/StakedTempleVoteToken.sol

```diff
@@ -0,0 +1,204 @@
+    }
+
+    /**
+     * @dev Destroys a `value` amount of tokens from `account`, deducting from
+     * the caller's allowance.
+     *
+     * See {ERC20-_burn} and {ERC20-allowance}.
+     *
+     * Requirements:
+     *
+     * - the caller must have allowance for ``accounts``'s tokens of at least
+     * `value`.
+     */
+    function burnFrom(address account, uint256 value) public virtual override(IStakedTempleVoteToken, ERC20Burnable) onlyAuthorized whenNotPaused {
+        _burn(account, value);
+    }
+
+    /**
+     * @dev See {IERC20-transferFrom}.
+     *
+     * Emits an {Approval} event indicating the updated allowance. This is not
+     * required by the EIP. See the note at the beginning of {ERC20}.
+     *
+     * NOTE: Does not update the allowance if the current allowance
+     * is the maximum `uint256`.
+     *
+     * Requirements:
+     *
+     * - `from` and `to` cannot be the zero address.
+     * - `from` must have a balance of at least `value`.
+     * - the caller must have allowance for ``from``'s tokens of at least
+     * `value`.
+     */
+    function transferFrom(address from, address to, uint256 value) public virtual override(ERC20, IERC20) whenNotPaused returns (bool) {
+        return super.transferFrom(from, to, value);
+    }
+
+    /**
+     * @dev See {IERC20-transfer}.
+     *
+     * Requirements:
+     *
+     * - `to` cannot be the zero address.
+     * - the caller must have a balance of at least `value`.
+     */
+    function transfer(address to, uint256 value) public virtual override(ERC20, IERC20) whenNotPaused returns (bool) {
+        return super.transfer(to, value);
+    }
+
+     /**
+     * @dev See {IERC20-approve}.
+     *
+     * NOTE: If `value` is the maximum `uint256`, the allowance is not updated on
+     * `transferFrom`. This is semantically equivalent to an infinite approval.
+     *
+     * Requirements:
+     *
+     * - `spender` cannot be the zero address.
+     */
+    function approve(address spender, uint256 value) public virtual override(ERC20, IERC20) whenNotPaused returns (bool) {
+        return super.approve(spender, value);
+    }
+
+    /**  
+     * @notice Get vote weight of an account
+     * @param account Account
+     */
+    function getVoteweight(address account) external override view returns (uint256) {
+        return ITempleGoldStaking(staking).getVoteweight(account);
+    }
+
+    /**  
+     * @notice Pause contract
+     */
+    function pause() external onlyElevatedAccess {
+        _pause();
+    }
+
+    /**  
+     * @notice Unpause contract
+     */
+    function unpause() external onlyElevatedAccess {
+        _unpause();
+    }
+
+    /**  
+     * @notice Move tokens from caller to dst, specific to Chief DAO contract. src is `msg.sender`
+     * @param dst Recipient
+     * @param amount Amount
+     */
+    function push(address dst, uint256 amount) external override onlyAuthorized {
+        transfer(dst, amount);
```

Comment:
I think this could just call the internal ERC20 `_transfer(owner, to, amount);` function directly, rather than using the public function?